test/image/mocks/zz-text_on_shapes_reversed_axes.json

```diff
@@ -0,0 +1,141 @@
+{
+    "data": [
+        {
+            "x": [
+                0,
+                50
+            ],
+            "y": [
+                0,
+                50
+            ]
+        }
+    ],
+    "layout": {
+        "width": 800,
+        "height": 600,
+        "margin": {
+            "t": 100,
+            "b": 50,
+            "l": 100,
+            "r": 50
+        },
+        "yaxis": {
+            "autorange": "reversed"
+        },
+        "template": {
+            "layout": {
+                "shapes": [
+                    {
+                        "label": {"text": "label"},
+                        "name": "myPath",
+                        "editable": true,
+                        "layer": "below",
+                        "line": {
+                            "width": 0
+                        },
+                        "fillcolor": "gray",
+                        "opacity": 0.5,
+                        "xref": "paper",
+                        "yref": "paper",
+                        "path": "M0.5,0.3C0.5,0.9 0.9,0.9 0.9,0.3C0.9,0.1 0.5,0.1 0.5,0.3ZM0.6,0.4C0.6,0.5 0.66,0.5 0.66,0.4ZM0.74,0.4C0.74,0.5 0.8,0.5 0.8,0.4ZM0.6,0.3C0.63,0.2 0.77,0.2 0.8,0.3Z"
+                    }
+                ]
+            }
+        },
+        "shapes": [
+            {
+                "label": {"text": "label 1 (top right)", "textposition": "top right"},
+                "editable": true,
+                "layer": "below",
+                "type": "rect",
+                "line": {
+                    "width": 5
+                },
+                "fillcolor": "red",
+                "opacity": 0.5,
+                "x0": 25,
+                "y0": 25,
+                "x1": 75,
+                "y1": 75
+            },
+            {
+                "label": {"text": "label 2"},
+                "editable": true,
+                "type": "circle",
+                "line": {
+                    "width": 5
+                },
+                "fillcolor": "green",
+                "opacity": 0.5,
+                "x0": 125,
+                "y0": 25,
+                "x1": 175,
+                "y1": 75
+            },
+            {
+                "label": {"text": "label 3"},
+                "editable": true,
```

Comment:
If you want these `editable` shapes to update their labels during drags, then please simply pick my commit from https://github.com/plotly/plotly.js/commit/d9acfab899cfe3eb15dcc6bc3dcc88c4c862c229 :)

Reply:
Amazing, done! Works perfectly. Thanks @archmoj !